src/sbvr-api/jobs.ts

```diff
@@ -0,0 +1,261 @@
+import * as nodeSchedule from 'node-schedule';
+import { AnyObject } from 'pinejs-client-core';
+
+import * as errors from './errors';
+import { addPureHook } from './hooks';
+import * as sbvrUtils from './sbvr-utils';
+import type * as Db from '../database-layer/db';
+import { permissions } from '../server-glue/module';
+
+const apiRoot = 'jobs';
+const DEFAULT_RETRY_LIMIT = 10;
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const modelText: string = require(`./${apiRoot}.sbvr`);
+
+interface Job {
+	id: number;
+	key: string;
+	is_executed_by__handler: string;
+	is_executed_with__parameter_set: any;
+	start_time: Date;
+	retry_limit: number;
+	error_count: number;
+	last_error?: string;
+	status: string;
+}
+interface JobHandler {
+	parameters: {
+		[key: string]: string;
+	};
+	callback: (parameterSet: any) => Promise<void>;
+}
+
+// TODO: Need to scope job handlers to model?
+const jobHandlers: {
+	[name: string]: JobHandler;
+} = {};
+
+/**
+ * Get and return actor from hook request object.
+ * @param req - Hook request object
+ * @returns Actor ID
+ * @throws BadRequestError if actor is missing
+ */
+function getActor(req: sbvrUtils.HookReq): number {
+	const actor = req.user?.actor ?? req.apiKey?.actor;
+	if (actor == null) {
+		throw new errors.BadRequestError(
+			'Scheduling job with missing actor on req is not allowed',
+		);
+	}
```

Comment:
@fisehara This currently always throws here in tests as `req.user` is always undefined. I've tried a number of things including setting `passthrough.req.user` on the pineTest client and `await null;` in the jobs preparse/postparse hooks. Any ideas?

Reply:
Nevermind, I did some digging and found out how to create and pass an apikey as part of the request through PineTest (which also works as my hooks checks for the actor from both `req.user` and `req.apiKey`).